### ISSUE TITLE
Reduce effectiveness of USA Airforce Point Defense Laser

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#638]
Explore feature for countering USA Airforce Point Defense Laser (PDL).

### Proposal 1
Add a new laser lock mode to USA Missile Defender, that works similar to traditional laser lock, but instead of sending damaging rockets it will just fire alot of dummy bullets to overwhelm the PDL ability of the targeted aircraft and/or vehicle. With such feature, USA can try to combine targeting and attacking Missile Defenders to combat against USA Airforce units. **Strengthens USA vs Airforce and Avengers.**

### Proposal 2
Increase the USA Airforce PDL reload time to reduce its effectiveness.
**> Weakens Airforce against all factions.**

### Proposal 3
Add PDL to SWG Chinooks, and maybe make them as cheap as AFG Chinooks too.
**> Strengthens SWG against all factions.**

### Proposal 4
Make USA Airforce Chinook and King Raptor PDL an upgrade from the Strategy Center.
**> Weakens Airforce against all factions in early game. Combat Chinook rushes and fast King Raptors are no longer as difficult to stop.**

### Proposal 5
Make Projectiles from Missile Defender immune to PDL.
**> Strengthens USA vs Airforce and Avengers.**

* #1106

### Proposal 6
King Raptor has two PDL modules assigned and is twice as affective as Chinook. Remove one of them and tweak the remainders desired effectiveness.
**> Weakens Airforce against all factions.**

* #1107

### Proposal 7
Make Projectiles from USA Missile Defender more resistant to PDL, so they take more zaps to be destroyed.
**> Strengthens USA against Airforce and Avengers**

* #1108

### Proposal 8
Make Projectiles from all Rocket Men more resistant to PDL, so they take more zaps to be destroyed.
**> Strengthens all factions against Airforce and Avengers**

### Proposal 9
Activate Aircraft PDL only after purchasing the Countermeasures Upgrade in the Airfield. Remove traditional countermeasures flares if required. Note: Countermeasures Upgrade does not apply Armor boost on King Raptor unlike normal Raptor.
**> Weakens Airforce against all factions in very early game. Combat Chinook rushes and fast King Raptors are no longer as difficult to stop.**

### Proposal 10
Make PDL a special ability like laser lock is a special ability of USA Missile Defenders. When PDL icon is pressed that unit activates its PDL ability for X seconds. After X seconds passed it will cooldown for Y seconds until it can be activated again.
**> Weakens Airforce against all factions. Increases skill requirements to master Airforce.**

### Proposal 11
Make PDL an unit upgrade like the Helix Gattling upgrade for a fair price. Applicable to Chinook, Combat Chinook or King Raptor or all of them.
**> Weakens Airforce against all factions.**

### Proposal 12
Have Projectiles from Missile Defender inflict more damage to USA Airforce Aircraft.
**> Strengthens USA vs Airforce.**

### Proposal 13 **APPLIED**
Remove 1 PDL Module from King Raptor and halve ReloadTime of the remaining one. This would allow a volley of two rockets instead of three rockets to break through the PDL onto the King Raptor.
**> Weakens Airforce against all factions.**

* #1555

### Proposal 14
Same as Proposal 10, but spawn a drone with PDL for the duration of the Ability.
**> Weakens Airforce against all factions. Increases skill requirements to master Airforce.**

**Comment by alanblack166** [comment]
There are no means by which the ``PointDefenseLaserUpdate`` module can be disabled without disabling the object itself. The most straightforward feature would be adding a weapon that fires extremely low damage rockets with a very high rate of fire, with the express purpose of overwhelming the PDL.

**Comment by xezon** [comment]
Interesting. Is there a way to target one specific unit only with such approach? Overwhelming PDL would mean that it also affects nearby units with PDL, and not just one. Whether that is good or bad, is difficult to say now.

**Comment by alanblack166** [comment]
It can be made to work similarly to Laser Lock - only one unit at a time. And yes, it would indeed affect other sources of PDL, unless we give this dummy projectile an exclusive KINDOF that is only valid for ``PointDefenseLaserUpdate`` module used by Airforce General's units:

```
  Behavior = PointDefenseLaserUpdate ModuleTag_Laser_One
    WeaponTemplate        = AirF_PointDefenseLaser
    PrimaryTargetTypes    = BALLISTIC_MISSILE SMALL_MISSILE SOME_OTHER_KINDOF
    ScanRate              = 0
    ScanRange             = 200.0
    PredictTargetVelocityFactor = 1.0
  End
  ```

**Comment by xezon** [comment]
That is good yes. This way we could have this ability work only for Chinook + King Raptor, but not for Avenger. I guess the overpowering approach is ok. It would also give visual confirmation by triggering PDL Zaps.

Would that be something you could implement a prototype for hanfield?

**Comment by alanblack166** [comment]
Yeah I could, shouldn't be too difficult.

**Comment by ZekeDlyoung** [comment]
Why not just reduce the King Raptors PDL delay between shots, or give it a clip?

**Comment by xezon** [comment]
That was done in Community Patch "1.06" and it was not so well received, because it changes PDL strength fundamentally.

**Comment by ZekeDlyoung** [comment]
And creating a specific weapon to deal with this specific issue for only one faction is better?

**Comment by xezon** [comment]
We don't know. Can be tested.

**Comment by ZekeDlyoung** [comment]
1.06's method can also be tested and adjusted, nobody said we should reduce the delay to the same amount as them

EDIT: It is the simplest and most elegant solution

**Comment by Jundiyy** [comment]
Whilst the new ability does bring out new micro possibilities, I'd prefer a reduction in speed time as Zeke has said (if we go down this route), tbh you don't need to reduce it a lot to make a bigger impact. Not saying it would be perfectly balanced by reducing the speed by 10%/20% but you'd see the difference quite quickly.

**Comment by xezon** [comment]
I added that as Proposal 2 on the opening post. Note that PDL rate change will affect all matchups and not just USA matches, which can be desirable or not.

**Comment by Jundiyy** [comment]
That is a true point.

**Comment by ZekeDlyoung** [comment]
IIRC, the King Raptor has flares as well, so it's not like making the laser a bit slower is gonna drop its performance _that_ sharply.

I could be wrong though

**Comment by xezon** [comment]
The issue with making PDL weaker is that strength is taken away, which is generally seen unfavourably as opposed to adding strengths.

**Comment by ExiLe-ZH** [comment]
One thing many ppl seem to forget is that PDL is a pure bonus strength, something air doesn't rely on, without it the faction would still perform statistically better in any matchup (except vs Tank) than the other USA's, because it has more pros like better Genpowers and strong Air units. 

Obviously we want to keep PDL's because there are fun and nostalgic factors involved. Reason why moving PDL's to SWG is probably not going to be accepted (I wouldn't either).

Imo this makes weakening PDL's an acceptable option, because in any form or shape it's still an advantage that other USA's don't have. I think it's about finding a balance where PDL's offer some decent protection but can't cancel out enormous volumes of rockets like currently the case (e.g. completely cancelling out Tech RPG harassment from GLA).

**Comment by ZekeDlyoung** [comment]
> Proposal 3
> Add PDL to SWG Chinooks, maybe make them as cheap as AFG Chinooks too.

Why is this a proposal? 

Also, if the main issue is America not being able to counter AirF Gen PDLs properly, perhaps one solution would be to make the Avenger available earlier albeit with its other abilities (PDL and laser pointer) locked behind a Strategy Center upgrade.

You know, maybe make some of the other existing units more viable, rather than jumping straight to adding new stuff.

Another, possibly unpopular definitely controversial, idea would be to replace the Humvee TOW missile upgrade with a laser cannon upgrade instead. The Humvee does have the voices for it from an unused Humvee Laser Cannon Upgrade for Townes. In this way the humvee can act as an early game non-missile AA weapon for USA and also potentially a counter for China ECM tanks.

**Comment by xezon** [comment]
The proposals were added because they were mentioned in a discussion thread on gamereplays.org

https://www.gamereplays.org/community/1.04+_Balance_Changes_Debate._Alternative_ideas-t1059356.html

**Comment by Stubbjax** [comment]
Proposal 1: Introducing a special ability for the sole purpose of countering a single unit is a complicated and messy approach to an otherwise simple problem. It seems as if the two abilities would be mechanically identical but required for different targets, which is unnecessary, counterintuitive and poor design. Would also be highly controversial.

Proposal 2: Should be to _increase_ the reload time. This is the most simple, effective and likely least controversial solution. It must be noted that this change would not affect all match-ups in the same way, and would primarily make the lives of other USAs (who mostly rely on rockets for anti-air) much easier. There is also a lot of flexibility here, such as increasing PDL reload time while also reducing damage received by rockets.

Proposal 3: It is not clean design to dish out arbitrary Air Force bonuses to other factions as compensation for those bonuses being too strong. This would be very controversial.

Proposal 4: Not a bad idea, but is a rather big change to to a problem that can be solved in much simpler and subtler ways. If an upgrade was going to be added, I'd suggest essentially replacing the Laser Missiles upgrade and having the upgrade in the airfield instead. Very controversial.

Proposal 5: This is not only more complicated (you'd need an entirely new missile type in addition to `BALLISTIC_MISSILE` and `SMALL_MISSILE`), but is also not clean design as it would be introducing a special case that contradicts well-established rules and game mechanics.

The main frustration for players is having King Raptors continuously flying care-free over their bases and hitting their chinooks / buildings with no effective counterplay. This is largely due to the lack of risk that King Raptors face against other USA factions. Players need at least half a dozen synchronised tow missile humvees just to hit one raptor, and missile defender rockets are too slow as raptors fly past. Reducing the PDL's rate of fire would increase the risk of flying over rocket-based weaponry, and would give opponents a more effective counter against them. It would also increase the amount of skill required of Air Force players in micromanaging their raptors' flight paths to avoid rocket-heavy areas, rather than simply targeting a chinook and forgetting about them, as currently happens.

The King Raptor has two PDL modules assigned to it in the game data, each with a 250ms reload time, as shown below. It would be a good idea to simply remove one of them and then tweak the other one accordingly. Even just removing one and halving the reload time of the other PDL would make a big difference, as it would reduce the raptor's ability to deflect groups of rockets while still maintaining the same deflection rate.

```
Behavior = PointDefenseLaserUpdate ModuleTag_22
  WeaponTemplate        = AirF_RaptorPointDefenseLaser
  PrimaryTargetTypes    = BALLISTIC_MISSILE SMALL_MISSILE
  ScanRate              = 10
  ScanRange             = 200.0
  PredictTargetVelocityFactor = 2.0
End
```
```
Behavior = PointDefenseLaserUpdate ModuleTag_Laser_One
  WeaponTemplate        = AirF_PointDefenseLaser
  PrimaryTargetTypes    = BALLISTIC_MISSILE SMALL_MISSILE
  ScanRate              = 0
  ScanRange             = 200.0
  PredictTargetVelocityFactor = 1.0
End
```

**Comment by ExiLe-ZH** [comment]
Besides firerate/reload time another proposal in the Discord is tweaking missile health values per missile type so that it takes multiple zaps to take out certain missiles.

**Comment by ZekeDlyoung** [comment]
Increasing MD missile health seems fine to me, though I still feel the KR's PDL should still be nerfed in some way, either by removing the second PDL, or slightly increasing their scan rate/reload time

**Comment by ExiLe-ZH** [comment]
Can you also reduce the PDL damage from the KR? Downside is that you immediately make them 50% worse vs every type of missile then?

**Comment by ReLaX82** [comment]
Make Projectiles from Missile Defender more resistant to PDL would my approach be.

Whatever you do, MDs shall be laserlock with this one button which is ingame.

**Comment by ReLaX82** [comment]
**Proposal 5**

Make Projectiles from Missile Defender immune to PDL. Strengthens USA vs Airforce.
Thought about this again: Best approach for me as it changes only 4 mus, AF vs all 4 USAs. Only issue is that AF-Mirror will massively changed.

**Comment by xezon** [comment]
Proposal 4 + Proposal 6 read interesting to me. King Raptors en masse are indistructible against rockets. Nerfing King Raptor PDL and moving PDL for both Chinook and Raptor to a new Upgrade will give all armies a better fighting chance against Airforce in both early and late game.

**Comment by xezon** [comment]
3 new proposals have been added.

**Comment by ZekeDlyoung** [comment]
If there is a way to make it work properly, Proposal 10 is good.

Otherwise, I'm still sticking with Proposal 2 + CMs armor upgrade for KRs.

Personally, I would go with an edited version of Proposal 9: Give AirF Gen CMs at the start (armor + flares), then replace his CMs upgrade with a PDL upgrade which further increases anti-missile defense and brings it back to OP ZH status

**Comment by xezon** [comment]
The risk of more armor for King Raptor is that Quads, Gattlings and Avengers will need longer to kill them, which can have undesired consequences in these battle scenarios. On the other hand, if missiles can help damage King Raptor more, then it may be ok. It is not clear to me.

**Comment by ZekeDlyoung** [comment]
Perhaps we can get some people to test it and see

**Comment by xezon** [comment]
# Discord talking points regarding Proposal 10

"Make PDL a special ability like laser lock"

PDL is one of the fundamental reasons why Airforce is overpowered, because rocket weapons make up a large portion of faction firepower. And Airforce can maneuver to be practically invincible against rockets.

## Original PDL setup
* Chinooks have 1 permanent PDL Module
* King Raptor has 2 permanent PDL Modules

## Proposed PDL Ability setup
* Chinooks have 1 PDL Module Ability
* King Raptor has 1 permanent PDL Module and 1 PDL Module Ability

### When PDL Ability is used
* Chinooks have 1 PDL Module active with identical effectiveness to Original
* King Raptor has 2 PDL Modules active with identical effectiveness to Original

### When PDL Ability is not used
* Chinooks have no PDL Module active
* King Raptor has 1 PDL Module active, resulting in half effectiveness to Original

### When can PDL Ability be used
* Always. When landed and flying.

### Duration and Cooldown
PDL Ability likely would need to last a few seconds and then cooldown a few seconds to become a utility for tactical timing. If the cooldown was too short, then it could just be spammed roboticly with hotkey and lose its meaning.

During cooldown, PDL is turned off. Prolonged PDL battles are no longer easily possible. For example King Raptors cannot simply circle above Stinger Sites zapping everything forever. Or King Raptors cannot be parked on Airfields while Rocket men shoot onto them. It would however be possible to enable PDL Ability on half the units in a group, and then enable the Ability on the other half while the first half cools down.

* Sample PDL Ability duration: 5s
* Sample PDL Ability cooldown: 5s

# Criticism

## [1] Five points raised by Stubbjax
> While it is certainly an interesting idea, I can absolutely guarantee the player base would not be happy if PDL was changed to an ability. I can see several immediate issues:

> 1. It breaks consistency with other units - Avengers, Paladins, (Stealth Fighters, Auroras, B2s) all automatically zap rockets, but there would then be a separate category of units (King Raptors, Chinooks) that do not automatically zap rockets, which is less intuitive and less graceful design-wise

> 2. It becomes unmanageable when there are more than a few aircraft

> 3. Air combat is often too fast to react in time

> 4. It would have a major impact on the current way in which the game is played (the biggest problem and where the highest degree of community ire would originate)

> 5. There are much more effective and simpler solutions. I'd suggest removing one PDL and halving the reload time of the other. At least as a start.

## [2] xezon answer to Stubbjax
> "absolutely guarantee the player base would not be happy if PDL was changed to an ability" is mostly meaningless, because there has been no such feature to experience in an actual match and accordingly there is zero data on how players perceived it.

> 1. I think it fits right into the current game and faction design. First, Comanche Helicopters have a Rocket Pods ability, Missile Defenders have a Laser Lock ability: Everyone uses these when possible and takes no issue with them. Second, Zero Hour has an abundance of Unit variants that are different. China Inf Outpost has 8 slots, China Outpost just 2, USA Chinook has no PDL, USA AFG Chinook has PDL. Visually many units are the same, but yet they have slight differences. This is just the way the factions are in Zero Hour. Third, PDL ability being different to PDL of Avenger & Co: Yes. Could give tzhe new PDL a different name variant and change the coloring of the lasers to be visually distinctive to the original permanent PDL module. I don't think game design is a problem at all. It fits.

> 2. This depends solely on skill of player. It comparable to Missile Defender Laser Lock: Some players can manage very incredible locks on enemy vehicles, while other players cannot. I think this is no problem at all.

> 3. True. There will be scenarios where it is difficult to react. You can compare it with a tunnel pop: If Plane attacks tunnel from fog, you may only have enough time to do a "Fargo Pop". Same principle applies to PDL ability. However, this is circumstancial and it is the players job to scroll, scout and react as best as he can. I think this is no problem at all.

> 4. For noob players who would not use abilities like Rocket Pods and Laser Lock and Co this would have a bigger impact than better players who do use abilities indeed. The goal of the PDL nerf is to give enemy factions a better fighting chance. And I think it would. I would even go so far to claim that a PDL nerf is the only big thing we need to do to Airforce to make all matchups involving AFG much more playable. USA vs Airforce will benefit greatly. USA can harrass supplies with Tow-Humvee or Rocket-Men. AFG needs to pull Chinooks away eventually. GLA vs Airforce will benefit. Rocket Men can harrass, Stinger Sites can defend, a Palace means something. And China, especially China Inf, will benefit. Hordes of rocket men, that Infantry General relies upon, will pose a much bigger threat to planes. This is good for this matchup. Nerfing PDL would nerf Combat Chinook Rush, rocket-invincible King Raptors and well protected AFG Supply Chinooks.

> 5. Simpler solution, absolutely. More effective on paper, I have not read them yet. You suggested to remove 1 PDL from King Raptor and halve effectiveness of the other. This is a vastly different change with vastly different effects. It permanently nerfes King Raptor (and Chinook if same principle applies) by reducing PDL effectiveness by 75%. It changes all unit dynamics at all times. I do agree your solution is much simpler to implement and communicate, but it is a vastly inferior change.

## [3] Stubbjax answer to xezon
> 1. I did not say the ability doesn't fit; I'm merely highlighting the fact that it is introducing an inconsistency to how all the other PDLs work, and how they have always worked. Not only that, but players would have to re-learn how to play with it, which is arguably worse than learning it for the first time

> 2. Sure, but it would still become unmanageable. Imagine if you had to manually command resource gatherers to collect supplies for every payload - you'd certainly end up with a greater skill cap. But that doesn't mean it's a good idea, for hopefully obvious reasons

> 3. It certainly would be difficult to react in a lot of situations; but the scope is limited due to the fast-paced nature of air combat, plus it adds an additional layer of complexity / responsibility / difficulty on an army that already has a fairly high skill ceiling that players simply don't need

> 4. I absolutely agree that the King Raptor's PDL is a primary balancing factor here - I'm not arguing against that at all. But the way you have worded this reads to me like you want to completely change the way Air Force is played or played against, which I strongly recommend against if widespread adoption is a goal of this project. Highly perceivable changes like the addition of an ability should be very carefully considered, and only done when absolutely necessary, if ever

> 5. Where did you get 75% from? One PDL at 125ms vs 2 PDLs at 250ms is not a 75% reduction. It's actually a 0% change in raw deflection rate (1 rocket per 125ms). The change would simply mean a volley of two rockets would break through a King Raptor's PDL instead of three. The key difference is that it reduces the raptor's ability to deflect groups of rockets, and gives opponents a greater avenue for counterplay by grouping volleys of rockets together. Saying it "permanently nerfs" the King Raptor is a bit insincere here. I also fail to understand how it is a 'vastly inferior change' - that just comes across as personal bias
Also, to be clear, I'm not saying the ability idea is bad at all. In fact, it is a very interesting and innovative idea that would probably find a great home in a mod. There are just far too many issues and implications with this approach and too much of a hard shift from the traditional gameplay for it to be a top contender

## [4] xezon answer to Stubbjax
> 1. Yes it is new feature essentially.

> 2. Whatever you think is obvious, is not. Just recently Dominator wrote Zero Hour is an easy game to play. Good players also unanimously welcome game mechanics that make them busy. Remember that Scrap deny bug where player puts scaffolds on top of scraps just to cancel them a few moments later? The pros like it. Not because it is a fascinating feature and not just because it helps China, but because it gives them more to do and sets them apart from lower level players. Or remember the ECM bug where it would auto target approaching vehicles instead of standing put to simply deflect missiles? Apparently some players like to micromanage the ECM around to counteract this bug. Is this attitude stupid? I don't think it is. Player want to do things in the game. And the more a player can do, the more he can achieve to master.

> 3. It has been widely agreed in community that Airforce General sits at the very top by a far margin with 71% win rate in 1v1. See: https://www.gamereplays.org/community/index.php?showtopic=965214 Airforce General is very capable to suck up something that makes it weaker or harder to master. If Airforce was so difficult to play with, how could it be that strong not just in Pro games, but also in Semi and Noob games? Do we need to assume that human players are not capable of using 1 new ability on Chinook and King Raptors, while they are perfectly capable to use them on other units?

> 4. Yes. All changes need to be evaluated in tests and regular matches.

> 5. I misread. It read like halving the effectiveness of the remaining PDL module. I understand now. It is valid suggestion.

## [5] Stubbjax answer to xezon
> 1. Which should really be reserved as a last resort when there are no better alternatives

> 2. You have to ask why a select few pros like it. Maybe because the alternative is worse? Maybe they can't imagine anything else? Maybe they want to maintain their edge over other players? When scrap denial is the only response you have to the overpowered rocket technical, of course you are going to abuse it. Plus, you're always going to find at least someone who enjoys a poorly designed feature (or in this case, oversight), but that does not necessarily give it any meaningful validation. There are no doubt players who love how overpowered Air Force is. If there's anything I've learnt as a game designer, it's that letting the pro players drive a game's design is incredibly unhealthy, and typically leads to the game's demise

> 3. I'm well aware of the balance issues and I certainly agree some kind of action is required. I'd also like to point out that making it harder to master does not directly address the balance issue. A perfectly played / mastered Air Force should not outmatch every other perfectly played / mastered faction

> 4. Not everything needs to be tested, or you will end up spending years on this project. Some ideas just clearly don't work

> 5. Great!

## [6] xezon answer to Stubbjax
> 1. If there are better alternatives we need to hear them.

> 2. I already told you why. Because they like having an edge and clicking things to make an impact on the match. The same reason someone clicks to have Missile Defender laser lock a Tank. Sure you can make game so that every rocket man auto laser locks everything, but then it just becomes much easier to use and you will see less deviation in Laser Lock battle outcomes. This is basically what happens with PDL. It is a free no brain extra ability that is always on and makes all units that have it much much better.

> 3. PDL Ability with cooldown would automatically weaken PDL over longer durations, because it is no longer permanent. It will be impossible to fly over Rocket Man with 1 Chinook indefinitely for example. However, the ability use with good timing preserves all original perks and preserves the pristine strength of AFG PDL. With PDL Ability it just could not be abused anymore the way it can be now (e.g. Combat Chinook Rush)

> 4. We need to push back against those who try to sell us bugs as features but we also need to push back against those that just make wild claims without evidence. If something already does not work in theory, then it must be proven so. For this particular case I have no yet been convinced that it would not help USA, China and GLA against Airforce.

**Comment by xezon** [comment]
2 new Proposals added.

**Comment by commy2** [comment]
I implemented proposals 5, 6 and 7.

I also tried to implement proposal 4. I was successful in making the PDL an upgrade for the King Raptor (and Stealth Fighter and Aurora), but I was unable to get it to work for Chinooks. This is because the implementation requires a HelixContain module. This module conflicts with the ChinookContain module. Having both overwrites one or the other. And without ChinookContain module, the Chinook can no longer collect supply or transport units.

This means that proposal 4, as well as 9 and 11 are not feasible without engine changes.

**Comment by ZekeDlyoung** [comment]
> I implemented proposals 5, 6 and 7.
> 
> I also tried to implement proposal 4. I was successful in making the PDL an upgrade for the King Raptor (and Stealth Fighter and Aurora), but I was unable to get it to work for Chinooks. This is because the implementation requires a HelixContain module. This module conflicts with the ChinookContain module. Having both overwrites one or the other. And without ChinookContain module, the Chinook can no longer collect supply or transport units.
> 
> This means that proposal 4, as well as 9 and 11 are not feasible without engine changes.

You can make the upgrade/ability spawn an invisible drone that has a wander distance of 1 to follow the chinook very closely and add the PDL there, that's what mods usually do.

**Comment by xezon** [comment]
Proposal 13 is applied.

**Comment by xezon** [comment]
So King Raptor is now a little bit easier to deal with by all factions.

Anything else we would like to do?

**Comment by MTKing4** [comment]
This is heavily reliant on the PDL testing results, but from reading through the proposals, i don't think we need anything else for now, we can go back if proposal 13 was not satisfactory.

**Comment by ReLaX82** [comment]
Wow, good work on number 13, well done.
Agree to test it!

**Comment by ExiLe-ZH** [comment]
Anyone available in 1 hour to test?

**Comment by ExiLe-ZH** [comment]
Today? :D

**Comment by xezon** [comment]
I suggest arrange tests in Discord. GitHub is not ideal for adhoc meet ups.
